### PR TITLE
GWL timing table

### PR DESCRIPTION
**Pull request by wkrantz**
# Description of PR

### Summary of changes and related issue
[What's changed in this PR?]

### Relevant motivation and context
[Why did you change this and what applicable context is needed to understand why this change is needed?]

### Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

<br>

#### Practical
- [ ] 80% unit test coverage
- [ ] Documentation
  - [ ] All functions/adjusted functions documented in the [readthedocs](https://climakitae.readthedocs.io/en/latest/).
  - [ ] Documentation is pushed
- [X] Complex code commented
- [X] Naming conventions followed
  - [ ] Helper functions hidden with `_` before the name
- [X] Context of function is clearly provided
  - [X] Intent of function is provided
  - [ ] How to test, so that it is not siloed on scientists and anyone can review
  - [X] Appropriate manual testing was completed
- [ ] Any notebooks known to utilize the affected functions are still working
- [X] Linting completed and resolved

#### Conceptual
- [X] Doesn't replicate existing functionality
- [X] Aligns with general coding standard of existing functions
- [X] Matches desired functinonality from users/scientists

### CODE REVIEW

**Comment by nicolejkeeney** [comment]
[GWL timing table](https://trello.com/c/ptQOOc4m/728-gwl-timing-table)

**Review by vicford — COMMENTED** [review]
This looks great @wkrantz. My one general comment is related to general consistency across AE as a whole. Here we have 1.5, 2.0, 2.5, and 3.0 as WL options. Our current WL code has 1.5, 2.0, 3.0, and 4.0. I am slightly inclined to include 2.5 as a user-facing option in our current WL code as we know it's going to be prominent for users, and for us to include 4.0 in this table. 

Thoughts @nicolejkeeney?

File: climakitae/util/generate_gwl_timing_table.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+                x_95 = np.nan
+                x_5 = np.nan
+                year_warmlevel_reached = np.nan
+
+            row.extend([x_5, year_warmlevel_reached, x_95])
+
+        wl_timing_df.loc[warmlevel] = row
+
+    wl_timing_df.to_csv("gwl_timing.csv")
+    write_csv_file(wl_timing_df, "data/gwl_timing_table.csv")
```
Comment (nicolejkeeney) [inline climakitae/util/generate_gwl_timing_table.py:91]: Why is the table saved twice? In two different locations, with two different names?

Comment (nicolejkeeney) [reply climakitae/util/generate_gwl_timing_table.py:91]: I assume we don't need to save a copy of the table in the home directory, and maybe the first save call can be deleted? ```wl_timing_df.to_csv("gwl_timing.csv")```

File: climakitae/util/generate_gwl_timing_table.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    ssp245_data = read_csv_file(ssp245_file, index_col="Year")
+    ssp370_data = read_csv_file(ssp370_file, index_col="Year")
+    ssp585_data = read_csv_file(ssp585_file, index_col="Year")
+
+    ssp_dict = {
+        # "SSP 1-1.9": ssp119_data, # skip this one scenario because it does not reach any warming levels
+        "SSP 1-2.6": ssp126_data,
+        "SSP 2-4.5": ssp245_data,
+        "SSP 3-7.0": ssp370_data,
+        "SSP 5-8.5": ssp585_data,
```
Comment (nicolejkeeney) [inline climakitae/util/generate_gwl_timing_table.py:43]: slightly inclined to not have spaces in column names; i.e. "SSP_1-2.6" (replace space with underscore) but am open to being disagreed with here

File: climakitae/util/generate_gwl_timing_table.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    }
+
+    # specify options
+    warmlevels = [1.5, 2.0, 2.5, 3.0]
+    cmip_t = np.arange(2015, 2100, 1)
+
+    # construct dataframe
+    cols = []
+    for ssp in ssp_dict.keys():
+        cols.extend([ssp + "_05", ssp + "_mean", ssp + "_95"])
```
Comment (nicolejkeeney) [inline climakitae/util/generate_gwl_timing_table.py:53]: thinking we may want to increase the readability of these column names; perhaps "_05_quant" (or some other better / more descriptive name) to show that its the 5th percentile; same for "_95" of course

File: climakitae/util/generate_gwl_timing_table.py (hunk 1 above)
Comment (wkrantz) [reply climakitae/util/generate_gwl_timing_table.py]: Done! Just an oversight leaving both of those lines in there.

File: climakitae/util/generate_gwl_timing_table.py (hunk 2 above)
Comment (wkrantz) [reply climakitae/util/generate_gwl_timing_table.py]: Changed in latest push

**Review by nicolejkeeney — APPROVED** [review]
Looks great, thanks!

**Comment by nicolejkeeney** [comment]
> This looks great @wkrantz. My one general comment is related to general consistency across AE as a whole. Here we have 1.5, 2.0, 2.5, and 3.0 as WL options. Our current WL code has 1.5, 2.0, 3.0, and 4.0. I am slightly inclined to include 2.5 as a user-facing option in our current WL code as we know it's going to be prominent for users, and for us to include 4.0 in this table.
> 
> Thoughts @nicolejkeeney?

@vicford Good point. Looks like @wkrantz added 2.5 since your comment! Thanks! :D

**Pull request merged**